tests/test_a11y.py

```diff
@@ -0,0 +1,111 @@
+"""Using Axe-core, scan the Kitchen Sink pages for accessibility violations."""
+
+import time
+from http.client import HTTPConnection
+from pathlib import Path
+from subprocess import PIPE, Popen
+from urllib.parse import urljoin
+
+import pytest
+from playwright.sync_api import Page
+
+from .utils.pretty_axe_results import pretty_axe_results
+
+# Important note: automated accessibility scans can only find a fraction of
+# potential accessibility issues.
+#
+# This test file scans pages from the Kitchen Sink examples with a JavaScript
+# library called Axe-core, which checks the page for accessibility violations,
+# such as places on the page with poor color contrast that would be hard for
+# people with low vision to see.
+#
+# Just because a page passes the scan with no accessibility violations does
+# *not* mean that it will be generally usable by a broad range of disabled
+# people. It just means that page is free of common testable accessibility
+# pitfalls.
+
+path_repo = Path(__file__).parent.parent
+path_docs_build = path_repo / "docs" / "_build" / "html"
+
+
+@pytest.fixture(scope="module")
+def url_base():
+    """Start local server on built docs and return the localhost URL as the base URL."""
+    # Use a port that is not commonly used during development or else you will
+    # force the developer to stop running their dev server in order to run the
+    # tests.
+    port = "8213"
+    host = "localhost"
+    url = f"http://{host}:{port}"
+
+    # Try starting the server
+    process = Popen(
+        ["python", "-m", "http.server", port, "--directory", path_docs_build],
+        stdout=PIPE,
+    )
+
+    # Try connecting to the server
+    retries = 5
+    while retries > 0:
+        conn = HTTPConnection(host, port)
+        try:
+            conn.request("HEAD", "/")
+            response = conn.getresponse()
+            if response is not None:
+                yield url
+                break
+        except ConnectionRefusedError:
+            time.sleep(1)
+            retries -= 1
+
+    # If the code above never yields a URL, then we were never able to connect
+    # to the server and retries == 0.
+    if not retries:
+        raise RuntimeError("Failed to start http server in 5 seconds")
+    else:
+        # Otherwise the server started and this fixture is done now and we clean
+        # up by stopping the server.
+        process.terminate()
+        process.wait()
+
+
+@pytest.mark.a11y
```

Comment:
I decided register this as a custom mark instead of using the `-k` pytest flag for string matching